_posts/2021-04-26-integrating-airflow-and-okta.md

```diff
@@ -0,0 +1,96 @@
+---
+layout: post
+title:  "Integrating Airflow with Okta"
+author: kuntalb
+tags:
+- okta
+- airflow
+- featured
+team: Core Platform
+---
+
+
+At Scribd we decided to pair Airflow with Okta. Earlier we were using LDAP for
+authentication. This write up will describe the journey of integrating Airflow
+with Okta from the earlier LDAP setup.
+
+
+## Prerequisite:
+1. Okta with [API Access Management](https://developer.okta.com/docs/concepts/api-access-management/) enabled.
+1. [Flask-AppBuilder
+   3.2.2](https://github.com/dpgaspar/Flask-AppBuilder/tree/v3.2.2). Official
+   Airflow repo has a
+   [constraint](https://github.com/apache/airflow/blob/master/setup.cfg#L97) on
+   `flask-appbuilder~=3.1,>=3.1.1`, so we might need to use a fork to get this
+   integration going.
+1. `sqlalchemy>=1.3.18, <1.4.0`
+1. [ ] `authlib==0.15.3`
+
+## Okta Setup
+
+![Sample Okta Setup](/post-images/2021-04-okta-airflow/sample-okta-setup.png)
+<font size="3"><center><i>Sample Okta Setup </i></center></font>
+
+1. Create an OIDC Web application. Give it a name and leave the values under the “Configure OpenID Connect” section empty.
+1. Make note of the Client ID and the Client Secret, as you will need them for configuring the airflow webserver.
+1. In the “Allowed Grant Types” section, make sure you check all of the boxes.
+1. For the Login redirect URIs field, you will enter: `https://your-airflow-url-goes-here.com/oauth-authorized/okta`
+1. For the Initiate login URI field, you will enter: `https://your-airflow-url-goes-here.com/login`
+
+## Airflow Configuration
+
+`conf/webserver_config.py`
+
+    AUTH_TYPE = AUTH_OAUTH
+    OAUTH_PROVIDERS = [
+    {'name': 'okta', 'icon': 'fa-circle-o',
+            'token_key': 'access_token',
+            'remote_app': {
+                'client_id': <<>>,
+                'client_secret': <<>>,
+                'api_base_url': 'https://<<okta_url>>/oauth2/v1/',
+                'client_kwargs': {
+                    'scope': 'openid profile email groups'
+                },
+                'access_token_url': 'https://<<okta_url>>/oauth2/v1/token',
+                'authorize_url': 'https://<<okta_url>>/oauth2/v1/authorize',
+        }
+        }
+    ]
+
+### Special Steps:
+
+
+1. We started with Flask-AppBuilder 3.2.1, however it had a bug that needs to
+   be fixed, we raised a [PR](https://github.com/dpgaspar/Flask-AppBuilder/pull/1589) for Flask-AppBuilder:
+   fix: load user info for oktaCLOSED  to resolve that issue. That PR got
```

Comment:
I don't understand what this is trying to say, perhaps the `fix: load user` snippet is a copy paste error?